backend/sirius-web-spring-collaborative/src/main/java/org/eclipse/sirius/web/spring/collaborative/projects/EditingContextEventProcessor.java

```diff
@@ -157,18 +157,31 @@ private Disposable setupChangeDescriptionSinkConsumer() {
     private void publishEvent(ChangeDescription changeDescription) {
         if (this.sink.currentSubscriberCount() > 0) {
             IInput input = changeDescription.getInput();
+            UUID transactionId = input.getId();
             if (input instanceof RenameRepresentationInput && ChangeKind.REPRESENTATION_RENAMING.equals(changeDescription.getKind())) {
                 String representationId = ((RenameRepresentationInput) input).getRepresentationId();
                 String newLabel = ((RenameRepresentationInput) input).getNewLabel();
-                EmitResult emitResult = this.sink.tryEmitNext(new RepresentationRenamedEventPayload(input.getId(), representationId, newLabel));
-                if (emitResult.isFailure()) {
-                    String pattern = "An error has occurred while emitting a RepresentationRenamedEventPayload: {}"; //$NON-NLS-1$
-                    this.logger.warn(pattern, emitResult);
+                this.tryEmitRepresentationRenamedEvent(transactionId, representationId, newLabel);
+            //when a diagram is renamed from the explorer, the input is of type RenameDiagramInput
```

Comment:
No need for that comment, it should probably be in the architecture documentation (which will be written) instead of there.